src/components/search/SearchComponent.tsx

```diff
@@ -2,48 +2,60 @@ import React, { useState } from "react";
 import SearchBar from "./SearchBar";
 import FilterComponent from "./filters/FilterComponent";
 import { useFetchData } from "../context/FetchDataContext";
+import { Button } from "@mui/material";
+import { useSearchParams } from "../context/SearchParamsContext";
 
 interface SearchParams {
-  onSearchParamsChange: (params: string | null) => void;
+    onSearchParamsChange: (params: string | null) => void;
 }
 
 const SearchComponent: React.FC<SearchParams> = ({ onSearchParamsChange }) => {
+    const { query, setQuery, setFilters, filters, setSearchPerformed } = useSearchParams();
 
     const [searchQuery, setSearchQuery] = useState<string>('');
-    const [filters, setFilters] = useState<string[]>([]);
 
     const { setUrlSearchParams } = useFetchData();
 
     const handleSearch = () => {
 
-        if (searchQuery.trim() === '' && filters.length === 0) {
+        if (query.trim() === '' && filters.length === 0) {
```

Comment:
Just a friendly reminder here:
`An empty string ( '' ), the number 0, null , NaN, a boolean false, and undefined variables are all “falsy”.`
This if check can be shortened as `if (!query.trim() && !filters.length)`. Using a shortened form is not always a better choice, but when you know for sure the returned type, you can use them to make your code look cleaner.